src/Microsoft.Azure.SignalR.Common/ServiceConnections/MultiEndpointServiceConnectionContainer.cs

```diff
@@ -118,6 +118,23 @@ public Task WriteAsync(ServiceMessage serviceMessage)
             return Task.WhenAll(routed.Select(s => Connections[s]).Select(s => s.WriteAsync(serviceMessage)));
         }
 
+        public Task WriteAndWaitForAckAsync(ServiceMessage serviceMessage)
+        {
+            if (_inner != null)
+            {
+                return _inner.WriteAndWaitForAckAsync(serviceMessage);
+            }
+
+            var routed = GetRoutedEndpoints(serviceMessage, _endpointManager.GetAvailableEndpoints()).ToArray();
+
+            if (routed.Length == 0)
+            {
+                throw new AzureSignalRNotConnectedException();
+            }
+
+            return Task.WhenAny(routed.Select(s => Connections[s]).Select(s => s.WriteAndWaitForAckAsync(serviceMessage)));
```

Comment:
WhenAll?

Reply:
`WhenAny` for `GroupJoin` and `GroupLeave` success, or `WhenAll` fail